queasars/minimum_eigensolvers/base/termination_criteria.py

```diff
@@ -109,3 +110,184 @@ def check_termination(
+    the last generation. The change is taken as the maximum of the absolute change of the expectation value of
+    the best individual from the last and current generation and the median Hausdorff distance between the
+    expectation values of the last and current generation. It is then set in relation to the median expectation
+    value of the last generation. This means this criterion terminates if the best individual
+    AND the population as a whole do not change sufficiently. This termination criterion can be configured to only
+    terminate if the change falls below the threshold value multiple times consecutively
+
+    :param minimum_relative_change: threshold value for the change in expectation value relative to the last
+        generation's median expectation value, below which this criterion chooses to terminate the optimization
+    :type minimum_relative_change: float
+    :param allowed_consecutive_violations: determines how often the threshold value can be violated consecutively
+        before this criterion chooses to terminate. If set to 0, this terminates the first time the change falls
+        below the threshold value. If set to 2 for example, this terminates the first time the change falls below
+        the threshold three consecutive times. Must be at least 0.
+    :type allowed_consecutive_violations: int
+    """
+
+    def __init__(self, minimum_relative_change: float, allowed_consecutive_violations: int):
+        super().__init__()
+        self._minimum_relative_change: float = minimum_relative_change
+        self._allowed_consecutive_violations: int = allowed_consecutive_violations
+        self._relative_change_history: list[float] = [
+            10 * self._minimum_relative_change for _ in range(0, self._allowed_consecutive_violations + 1)
+        ]
+        self._last_population_evaluation: Optional[BasePopulationEvaluationResult] = None
+
+    def reset_state(self) -> None:
+        self._relative_change_history = [
+            10 * self._minimum_relative_change for _ in range(0, self._allowed_consecutive_violations + 1)
+        ]
+        self._last_population_evaluation = None
+
+    def check_termination(
+        self,
+        population_evaluation: BasePopulationEvaluationResult,
+        best_individual: BaseIndividual,
+        best_expectation_value: float,
+    ) -> bool:
+
+        if self._last_population_evaluation is not None:
+            hausdorff_distance: float = _median_hausdorff_distance_by_expectation_value(
+                self._last_population_evaluation, population_evaluation
+            )
+            best_individual_distance: float = abs(
+                self._last_population_evaluation.best_expectation_value - population_evaluation.best_expectation_value
+            )
+            distance: float = max(hausdorff_distance, best_individual_distance)
+            last_population_median_expectation: float = cast(
```

Comment:
It doesn't really matter performwise, but you could also keep this value in a field the same way as `_last_population_evaluation`. I would think is is a tiny bit more intuitive but I don't care much either way.

Reply:
Since it is not super important to you either way, i will just leave it as is.